src/function/matrix/subset.js

```diff
@@ -1,14 +1,14 @@
 import { isIndex } from '../../utils/is.js'
 import { clone } from '../../utils/object.js'
-import { validateIndex } from '../../utils/array.js'
+import { isEmptyIndex, validateIndex, validateIndexSourceSize } from '../../utils/array.js'
 import { getSafeProperty, setSafeProperty } from '../../utils/customs.js'
 import { DimensionError } from '../../error/DimensionError.js'
 import { factory } from '../../utils/factory.js'
 
 const name = 'subset'
-const dependencies = ['typed', 'matrix']
+const dependencies = ['typed', 'matrix', 'config', 'zeros', 'add']
 
-export const createSubset = /* #__PURE__ */ factory(name, dependencies, ({ typed, matrix }) => {
+export const createSubset = /* #__PURE__ */ factory(name, dependencies, ({ typed, matrix, config, zeros, add }) => {
   /**
    * Get or set a subset of a matrix or string.
    *
```

Comment:
Can you write an example of using the booleans and filtering here?

Can you also write a section about these new features on the documentation page about matrices (explaining both the usage in the expression parser as well as using the JS API)? See https://github.com/josdejong/mathjs/blob/develop/docs/datatypes/matrices.md

Reply:
Thanks, the docs are very well written 👌